application-analytics-ui/src/main/resources/Analytics/Code/Macros/ExitPagesJSON.xml

```diff
@@ -0,0 +1,119 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="Analytics.Code.Macros.ExitPagesJSON" locale="">
+  <web>Analytics.Code.Macros</web>
+  <name>ExitPagesJSON</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>ExitPagesJSON</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{include reference='Analytics.Code.Macros.VelocityMacros'/}}
+
+{{velocity}}
+#if ($xcontext.action == 'get')
+  #set ($offset = $numbertool.toNumber($request.offset).intValue())
+  ## The offset sent by the live table starts at 1.
+  #set ($offset = $offset - 1)
+  #if (!$offset || $offset &lt; 0)
+    #set ($offset = 0)
+  #end
+  #set ($limit = $numbertool.toNumber($request.limit).intValue())
+  #if (!$limit)
+    #set ($limit = 15)
+  #end
+
+  ## Mapping between the livedata columns and the Matomo fields.
+  #set ($matomoKeysMapping = {
+    'pageTitle' : 'label',
+    'visits': 'nb_visits',
+    'exitRate' : 'exit_rate',
+    'exitNbVisits':'exit_nb_visits'
+  })
+  #set ($sort = 'nb_hits')
+  #set ($order = 'desc')
+  #if ($request.sort)
+    #set ($sort = $matomoKeysMapping.get($request.sort))
+    #set ($order = $request.dir)
+  #end
+
+  #set ($filters = {})
+  #foreach ($key in $matomoKeysMapping.keySet())
+    #set ($value = $request.get($key))
+    #if ($value &amp;&amp; $value != '')
+      #set ($discard = $filters.put($matomoKeysMapping.get($key), $request.get($key)))
+    #end
+  #end
```

Comment:
I've seen this code too on ``entryPages``. Could be moved also to a Velocity macro.